lib/internal/make-api-call.js

```diff
@@ -75,12 +75,17 @@ exports.inject = function(options) {
     var useClientId = query.supportsClientId && clientId && clientSecret;
     delete query.supportsClientId;
 
-    var requestUrl = formatRequestUrl(path, query, useClientId);
+    var method = query.method;
+    delete query.method;
+
+    var isPost = method == 'POST'
+    var postData = isPost ? query : null;
+    var requestUrl = formatRequestUrl(path, query, method, useClientId);
 
     function rateLimitedGet() {
       return requestQueue.add(function() {
         return Task.start(function(resolve, reject) {
-          return makeUrlRequest(requestUrl, resolve, reject);
+          return makeUrlRequest(requestUrl, resolve, reject, postData);
```

Comment:
It worries me a little that we only support GET and POST, but never check that `method` isn't anything else.

Should `makeUrlRequest` take the method, as well as the postData ?